### PR TITLE
Ensure image permission of 755 when overriding the image

### LINKED ISSUE #4177: Overriding an image will not restore the correct permission
## Version of Singularity:

```
3.3
```

### Expected behavior

The correct permission is restored on the image.

### Actual behavior

```bash
$ touch	alpine_latest.sif
$ ls -l alpine_latest.sif
-rw-r--r-- 1 westleyk westleyk 0 Aug 10 08:52 alpine_latest.sif

$ singularity pull -F library://alpine                     
INFO:    Container is trusted - run 'singularity key list' to list your trusted keys
INFO:    Download complete: alpine_latest.sif

$ ls -l alpine_latest.sif
-rw-r--r-- 1 westleyk westleyk 2720699 Aug 10 08:53 alpine_latest.sif
```

The container permission is supposed to be

```
$ ls -l alpine_latest.sif 
-rwxr-xr-x 1 westleyk westleyk 2720699 Aug 10 08:54 alpine_latest.sif
```

This seems to be a bug since 3.0.0, but not 2.X.

### Steps to reproduce behavior

Write here.

### PR DESCRIPTION
**Pull request by WestleyK**
## Description of the Pull Request (PR):

This PR ensures that the image file permission is 755 when overriding the file.

### This fixes or addresses the following GitHub issues:

- Fixes #4177

<br>

### CODE REVIEW

**Review by gvallee — APPROVED** [review]
lgtm

File: internal/pkg/library/library.go [hunk 1]
```diff
@@ -123,3 +123,24 @@ func SearchLibrary(ctx context.Context, c *client.Client, value string) error {
...
+	// Perms are 777 *prior* to umask in order to allow image to be
+	// executed with its leading shebang like a script
+	destFile, err := os.OpenFile(path, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0777)
+	if err != nil {
+		return nil, fmt.Errorf("while opening destination file: %s", err)
+	}
+
+	// Ensure the file permission is 777. If path already exist, os.OpenFile
+	// will **not** apply the specified permission of 0777.
+	err = destFile.Chmod(0777)
```
Comment (jmstover) [inline internal/pkg/library/library.go]: Change this to `destFile.Chmod(0755)` ...

**Review by jmstover — CHANGES_REQUESTED** [review]
Don't set permission to `0777` ... World/Group writable not good.

File: internal/pkg/library/library.go (hunk 1 above)
Comment (jmstover) [reply internal/pkg/library/library.go]: Also ... this looks like it's going to override the users `umask` setting... Is that correct?

**Review by jmstover — APPROVED** [review]

**Review by ikaneshiro — APPROVED** [review]

**Pull request merged**